conan/tools/system/package_manager.py

```diff
@@ -206,21 +222,33 @@ def _update(self):
             return self._conanfile_run(command, self.accepted_update_codes)
 
     def _check(self, packages, host_package=True):
-        missing = [pkg for pkg in packages if self.check_package(self.get_package_name(pkg, host_package=host_package)) != 0]
+        missing = [pkg for pkg in packages if self.check_package(pkg, host_package) != 0]
         return missing
 
-    def check_package(self, package):
-        command = self.check_command.format(tool=self.tool_name,
-                                            package=package)
+    def check_package(self, package, host_package=True):
+        name, version, arch_separator, arch_name, _ = self._split_package_name(package, host_package)
+        package = self.full_package_name.format(name=name,
+                                                arch_separator=arch_separator,
+                                                arch_name=arch_name,
+                                                version="",
+                                                version_separator="")
+        if version and self.check_version_command:
```

Comment:
Maybe we should we output a warning message when we are specifying versions but the package manager does not support that?